pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -54,7 +55,45 @@ func NewController() *Controller {
 func (c *Controller) AddRegistry(registry Registry) {
 	c.storeLock.Lock()
 	defer c.storeLock.Unlock()
-	c.registries = append(c.registries, registry)
+	registries := c.registries
+	registries = append(registries, registry)
+	c.registries = registries
+}
+
+// DeleteRegistry deletes specified registry from the aggregated controller
+func (c *Controller) DeleteRegistry(registry Registry) {
+	c.storeLock.Lock()
+	defer c.storeLock.Unlock()
+	registries := c.registries
+	if len(c.registries) == 0 {
+		log.Warnf("Registry list is empty, nothing to delete")
+		return
+	}
+	index, ok := c.GetRegistryIndex(registry)
+	if !ok {
+		log.Warnf("Registry is not found in the registries list, nothing to delete")
+		return
+	}
+	registries = append(registries[:index], registries[index+1:]...)
+	c.registries = registries
+}
+
+// GetRegistries returns a copy of all registries
+func (c *Controller) GetRegistries() []Registry {
+	c.storeLock.Lock()
+	defer c.storeLock.Unlock()
+	registries := c.registries
+	return registries
+}
+
+// GetRegistryIndex returns the index of a registry
+func (c *Controller) GetRegistryIndex(registry Registry) (int, bool) {
+	for i, r := range c.registries {
+		if reflect.DeepEqual(registry, r) {
```

Comment:
Each registry has a unique name, right?

Reply:
SInce it is not a map, there is no guarantee about being unique, in theory they should though. I think it would be safer to compare with DeepEqual